var/spack/repos/builtin/packages/rust/package.py

```diff
@@ -103,10 +103,17 @@ class Rust(Package):
     phases = ["configure", "build", "install"]
 
     @classmethod
-    def determine_version(csl, exe):
-        output = Executable(exe)("--version", output=str, error=str)
-        match = re.match(r"rustc (\S+)", output)
-        return match.group(1) if match else None
+    def determine_spec_details(cls, prefix, exes_in_prefix):
+        rustc_candidates = list(x for x in exes_in_prefix if os.path.basename(x) == "rustc")
+        cargo_candidates = list(x for x in exes_in_prefix if os.path.basename(x) == "cargo")
+        # Both rustc and cargo must be present
+        if not rustc_candidates or not cargo_candidates:
+            return
```

Comment:
```suggestion
        if not (rustc_candidates and cargo_candidates):
            return
```
I might suggest a slight swap here to make it easier to read.